### DISCUSSION TITLE
Context?

### DISCUSSION THREAD

**Discussion by elgs** [redis/go-redis#2149] [General]
What is the reason the context.Context has to be exposed to the api?

I don't know how it's useful. But assuming it's useful, why is it not hidden in the implementations? This looks like a faulty api design.

**Comment by vmihailenco** [comment]
context.Context is used for [tracing](https://opentelemetry.uptrace.dev/guide/go-tracing.html#context) and [timeouts](https://dev.to/hgsgtk/timeout-using-context-package-in-go-1b3c).

**Comment by elgs** [comment]
Is it used by the client or the api implementation?

**Comment by elgs** [comment]
I looked into the context.Context and I wonder why not put the context as a member variable of the Client struct, so the context could be retrieved by `rds.Context()`, instead of passing through all methods?
